java-engine/src/main/java/io/getunleash/engine/CustomStrategies.java

```diff
@@ -0,0 +1,151 @@
+package io.getunleash.engine;
+
+import static java.util.function.Function.identity;
+import static java.util.stream.Collectors.toMap;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+class CustomStrategies {
```

Comment:
This name is a little off to my eyes. This doesn't tell me a lot about what it does. It especially looks weird for a class called CustomStrategies to take a parameter called customStrategies in its constructor.

CustomStrategyHandler? Dunno if I like that but it tells me more about what this does

Reply:
Yes, I carried this over from .Net, I was unsure about this name. I like your proposal but let me do some thinking before stamping that name :D